components/match2/wikis/heroes/match_group_input_custom.lua

```diff
@@ -0,0 +1,660 @@
+---
+-- @Liquipedia
+-- wiki=heroes
+-- page=Module:MatchGroup/Input/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local Json = require('Module:Json')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local String = require('Module:StringUtils')
+local Table = require('Module:Table')
+local Variables = require('Module:Variables')
+local ChampionNames = mw.loadData('Module:HeroNames')
+local Streams = require('Module:Links/Stream')
+
+local MatchGroupInput = Lua.import('Module:MatchGroup/Input', {requireDevIfEnabled = true})
+local Opponent = Lua.import('Module:Opponent', {requireDevIfEnabled = true})
+
+local _STATUS_SCORE = 'S'
+local _STATUS_DRAW = 'D'
+local _STATUS_DEFAULT_WIN = 'W'
+local _STATUS_FORFEIT = 'FF'
+local _STATUS_DISQUALIFIED = 'DQ'
+local _STATUS_DEFAULT_LOSS = 'L'
+local _ALLOWED_STATUSES = {
+	_STATUS_DRAW,
+	_STATUS_DEFAULT_WIN,
+	_STATUS_FORFEIT,
+	_STATUS_DISQUALIFIED,
+	_STATUS_DEFAULT_LOSS,
+}
+local ALLOWED_VETOES = {'decider', 'pick', 'ban', 'defaultban'}
+local _MAX_NUM_OPPONENTS = 2
+local _MAX_NUM_PLAYERS = 5
+local _DEFAULT_BESTOF = 3
+local _DEFAULT_MODE = 'team'
+local _NO_SCORE = -99
+local _DUMMY_MAP = 'default'
+local _NP_STATUSES = {'skip', 'np', 'canceled', 'cancelled'}
+local _EPOCH_TIME = '1970-01-01 00:00:00'
+local _DEFAULT_RESULT_TYPE = 'default'
+local _NOT_PLAYED_SCORE = -1
+local _NO_WINNER = -1
+local _SECONDS_UNTIL_FINISHED_EXACT = 30800
+local _SECONDS_UNTIL_FINISHED_NOT_EXACT = 86400
+
+-- containers for process helper functions
+local matchFunctions = {}
+local mapFunctions = {}
+local opponentFunctions = {}
+
+local CustomMatchGroupInput = {}
+
+-- called from Module:MatchGroup
+function CustomMatchGroupInput.processMatch(match, options)
+	-- Count number of maps, check for empty maps to remove, and automatically count score
+	match = matchFunctions.getBestOf(match)
+	match = matchFunctions.getScoreFromMapWinners(match)
+
+	-- process match
+	Table.mergeInto(
+		match,
+		matchFunctions.readDate(match)
+	)
+	match = matchFunctions.getOpponents(match)
+	match = matchFunctions.getTournamentVars(match)
+	match = matchFunctions.getVodStuff(match)
+	match = matchFunctions.getExtraData(match)
+
+	-- Adjust map data, especially set participants data
+	match = matchFunctions.adjustMapData(match)
+
+	return match
+end
+
+function matchFunctions.adjustMapData(match)
+	local opponents = {}
+	for opponentIndex = 1, _MAX_NUM_OPPONENTS do
+		opponents[opponentIndex] = match['opponent' .. opponentIndex]
+	end
+	local mapIndex = 1
+	while match['map'..mapIndex] do
+		match['map'..mapIndex] = mapFunctions.getParticipants(match['map'..mapIndex], opponents)
+		mapIndex = mapIndex + 1
+	end
```

Comment:
```suggestion
	for _, map in Table.iter.pairsByPrefix(match, 'map') do
		map = mapFunctions.getParticipants(map, opponents)
	end
```

Reply:
got **value assigned to variable 'map' is unused** on  `map = mapFunctions.getParticipants(map, opponents)`
do I need to remove both map,opponents within the ()?

Reply:
i don't get that warning in vscode